dialects/linux/dproc.c

```diff
@@ -1518,12 +1574,91 @@ process_proc_map(p, s, ss)
 		efs = sv = 1;
 	    else
 		efs = 0;
-	    if (!efs) {
-		if (HasNFS)
-		    sv = statsafely(fp[6], &sb);
-		else
-		    sv = stat(fp[6], &sb);
+
+	    /* kernel version >= 3.8.0 and the target process has a different
+	     * mount namespace with this lsof process. */
+	    if (compare_kernel_release(38000) && compare_namespace("mnt")) {
+		/* Fork a child process and create a pipe. */
+                if (pipe(pipefd) < 0) {
+                    fprintf(stderr, "%s: can't open pipes: %s\n",
+                        Pn, strerror(errno));
+                    goto stat_directly;
+                }
+
+                pid = fork();
+                if (pid < 0) {
+                    fprintf(stderr, "%s: can't fork: %s\n",
+                        Pn, strerror(errno));
+                    goto stat_directly;
+                }
+
+                if (pid == 0) {
+	            /* Child process:
+	             * Move this child process into the mount namespace of the
+	             * target process before applying stat(2).
+	             */
+                    int nsfd;
+                    char nspath[NS_PATH_LENGTH];
+
+                    close(pipefd[0]);
+                    sprintf(nspath, "/proc/%d/ns/mnt", Lp->pid);
+                    nsfd = open(nspath, O_RDONLY);
+
+                    if (nsfd < 0 || setns(nsfd, CLONE_NEWNS)) {
+                        fprintf(stderr, "%s: can't setns: %s\n",
+                            Pn, strerror(errno));
+			Exit(LSOF_ERROR);
```

Comment:
Unaligned on my web browser.

Reply:
Do we have to report this?
The original 'stat: No such file or directory' looks enough.

Reply:
> Do we have to report this? The original 'stat: No such file or directory' looks enough.

I think print error message can help to analyze why 'stat: No such file or directory' happens.